<a href="https://colab.research.google.com/github/shashwat0070/Operational_Research_ShashwatMishra_R103219017/blob/main/25_Aug_Exp_2_3_Shashwat_R103219017.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

OR LAB EXPERIMENT - 2,3
SHASHWAT MISHRA - 500075273 - R103219017 - CSE - BAO (5th Sem)

ANS-1

In [1]:
from scipy.optimize import linprog


In [2]:
obj = [-40 , -35]
lhs_ineq = [[2,3] , [4,3] ]
rhs_ineq = [60,96]
bnd = [(0, float("inf")), (0, float("inf"))]
opt = linprog(c=obj, A_ub=lhs_ineq, b_ub=rhs_ineq, bounds=bnd,method="revised simplex")
opt

     con: array([], dtype=float64)
     fun: -1000.0
 message: 'Optimization terminated successfully.'
     nit: 2
   slack: array([0., 0.])
  status: 0
 success: True
       x: array([18.,  8.])

Firm can earn Maximum Profit = Rs. 1000 when A = 18 and B = 8

ANS-2:

In [3]:
obj = [40 , 24]
lhs_ineq = [[-20,-50] , [-80,-50] ]
rhs_ineq = [-4800,-7200]
bnd = [(0, float("inf")), (0, float("inf"))]
opt = linprog(c=obj, A_ub=lhs_ineq, b_ub=rhs_ineq, bounds=bnd,method="revised simplex")
opt

     con: array([], dtype=float64)
     fun: 3456.0
 message: 'Optimization terminated successfully.'
     nit: 3
   slack: array([2400.,    0.])
  status: 0
 success: True
       x: array([  0., 144.])

Cost is Minimum i.e. Rs. 3456.0 when Bags of Mixture A = 0 and Bags of Mixture B = 144



ANS-3


In [4]:
obj = [-5 , -2]
lhs_ineq = [[10,2] , [1,0.5]]
rhs_ineq = [2100,600]
bnd = [(0, float("inf")), (800, float("inf"))]
opt = linprog(c=obj, A_ub=lhs_ineq, b_ub=rhs_ineq, bounds=bnd,method="revised simplex")
opt

     con: array([], dtype=float64)
     fun: -2100.0
 message: 'Optimization terminated successfully.'
     nit: 2
   slack: array([ 0., 75.])
  status: 0
 success: True
       x: array([   0., 1050.])

Profit is Maximum i.e. Rs. 2100 when No. of Units of A = 0 and No. of B = 1050 per week

ANS-4:

In [5]:
obj = [-60 , -40]
lhs_ineq = [[2,1] ]
rhs_ineq = [60]
bnd = [(0, 25), (0, 35)]
opt = linprog(c=obj, A_ub=lhs_ineq, b_ub=rhs_ineq, bounds=bnd,method="revised simplex")
opt

     con: array([], dtype=float64)
     fun: -2150.0
 message: 'Optimization terminated successfully.'
     nit: 3
   slack: array([0.])
  status: 0
 success: True
       x: array([12.5, 35. ])

Profit is Maximum i.e. Rs. 2150 when No. of Refridgerator = 12.5 and Ranges = 35

ANS-5

In [6]:
obj = [20 , 10]
lhs_ineq = [[1,2] , [-3,-1], [-4,-3] ]
rhs_ineq = [40,-30,-60]
bnd = [(0, float("inf")), (0, float("inf"))]
opt = linprog(c=obj, A_ub=lhs_ineq, b_ub=rhs_ineq, bounds=bnd,method="revised simplex")
opt

     con: array([], dtype=float64)
     fun: 240.0
 message: 'Optimization terminated successfully.'
     nit: 2
   slack: array([10.,  0.,  0.])
  status: 0
 success: True
       x: array([ 6., 12.])

Z = 240 , X1 = 6 and X2 = 12

ANS-6

In [7]:
obj = [-1 , -1 , -3]
lhs_ineq = [[3,2,1] , [2,1,2]]
rhs_ineq = [300,200]
bnd = [(0, float("inf")), (0, float("inf")), (0, float("inf"))]
opt = linprog(c=obj, A_ub=lhs_ineq, b_ub=rhs_ineq, bounds=bnd,method="revised simplex")
opt

     con: array([], dtype=float64)
     fun: -300.0
 message: 'Optimization terminated successfully.'
     nit: 1
   slack: array([200.,   0.])
  status: 0
 success: True
       x: array([  0.,   0., 100.])

Z = 300 , X1 = 0 , X2 = 0 , X3 = 100